pkg/scheduler/plugins/elastic/elastic.go

```diff
@@ -53,4 +53,24 @@ func JobOrderFn(l, r interface{}) int {
 	return 0
 }
 
+func minAvailableState(pgi *podgroup_info.PodGroupInfo) (bool, bool, bool) {
+	atLeastOneSubgroupUnderMinAvailable := false
+	allSubgroupsAboveMinAvailable := true
+	allSubgroupsExactlyMinAvailable := true
+
+	for _, subGroup := range pgi.SubGroups {
+		allocatedCount := int32(subGroup.GetNumActiveAllocatedTasks())
+		if allocatedCount < subGroup.GetMinAvailable() {
+			atLeastOneSubgroupUnderMinAvailable = true
+			allSubgroupsAboveMinAvailable = false
+		}
+		if allocatedCount == subGroup.GetMinAvailable() {
+			allSubgroupsAboveMinAvailable = false
+		} else {
+			allSubgroupsExactlyMinAvailable = false
+		}
+	}
+	return atLeastOneSubgroupUnderMinAvailable, allSubgroupsAboveMinAvailable, allSubgroupsExactlyMinAvailable
+}
```

Comment:
In this commit we generalize the use case of elastic comparison, but not handled all cases.

All previous cases are handled the same, meaning not breaking changes compared to the days when SubGroup were not exists - As the behavior of no SubGroups is equal to single default SubGroup.

The unhandled behavior is when we have multiple SubGroups in both jobs.
In each one subGroup is above MinMember and second is under MinMember.
Who is first?
Not handled